lib/src/test/kotlin/app/cash/nostrino/client/RelayClientTest.kt

```diff
@@ -153,6 +155,41 @@ class RelayClientTest : StringSpec({
     }
   }
 
+  "can consume events by hashtag" {
+    val sec = SecKeyGenerator().generate()
+    val noteWithoutHashTags = arbTextNote.next()
+    val noteWithHashTags = TextNote(
+      text = "never",
+      tags = listOf(
+        HashTag("gonna"),
+        HashTag("give"),
+        HashTag("you"),
+        HashTag("up")
+      )
+    )
+
+    with(RelayClient("ws://localhost:7707")) {
+      start()
+      subscribe(
+        Filter.hashTagNotes(
+          hashtags = setOf(
+            HashTag("give"),
+            HashTag("up"),
+          )
+        )
+      )
+
+      send(noteWithHashTags.sign(sec))
+      send(noteWithoutHashTags.sign(sec))
+
+      val actualEvent = notes.first()
+      actualEvent.content shouldBe "never"
+      actualEvent.content() shouldBe noteWithHashTags
+
+      stop()
```

Comment:
Should we assert the length of `notes` as well, to ensure we never received the note without hash tags?